x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DataTypeConverter.java

```diff
@@ -537,6 +548,10 @@ public static Object convert(Object value, DataType dataType) {
                 throw new QlIllegalArgumentException("[" + o + "] is not a valid IPv4 or IPv6 address");
             }
             return o;
+        }),
+        STRING_TO_VERSION(o -> {
+            // TODO validate...?
```

Comment:
Not sure this is needed. The underlying implementation seems to do no validation on VERSION fields